Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/DeleteOrphanManifestNodesTask.java

```diff
@@ -0,0 +1,116 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.experimental.autoingest;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.Level;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.coordinationservice.CoordinationService;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.progress.ProgressIndicator;
+
+/**
+ * A task class for cleaning up auto ingest job coordination service nodes for
+ * which there is no longer a corresponding manifest file.
+ */
+final class DeleteOrphanManifestNodesTask implements Runnable {
+
+    private static final Logger logger = Logger.getLogger(DeleteOrphanManifestNodesTask.class.getName());
+    private final ProgressIndicator progress;
+
+    /**
+     * Constucts an instance of a task for cleaning up case coordination service
+     * nodes for which there is no longer a corresponding case.
+     *
+     * @param progress
+     */
+    DeleteOrphanManifestNodesTask(ProgressIndicator progress) {
+        this.progress = progress;
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "ManifestNodesCleanupTask.progress.startMessage=Starting orphaned manifest file znode cleanup...",
+        "ManifestNodesCleanupTask.progress.connectingToCoordSvc=Connecting to the coordination service...",
+        "ManifestNodesCleanupTask.progress.gettingCaseNodesListing=Querying coordination service for manifest file znodes...",
+        "# {0} - node path", "ManifestNodesCleanupTask.progress.deletingOrphanedManifestNode=Deleting orphaned manifest file node {0}..."
+    })
+    public void run() {
+        progress.start(Bundle.ManifestNodesCleanupTask_progress_startMessage());
+        try {
+            progress.progress(Bundle.ManifestNodesCleanupTask_progress_connectingToCoordSvc());
+            logger.log(Level.INFO, "Connecting to the coordination service for orphan manifest file node clean up");  // NON-NLS            
+            CoordinationService coordinationService;
+            try {
+                coordinationService = CoordinationService.getInstance();
+            } catch (CoordinationService.CoordinationServiceException ex) {
+                logger.log(Level.WARNING, "Error connecting to the coordination service", ex); // NON-NLS
+                return;
+            }
+
+            progress.progress(Bundle.ManifestNodesCleanupTask_progress_gettingCaseNodesListing());
+            logger.log(Level.INFO, "Querying coordination service for case nodes for orphaned case node clean up");  // NON-NLS
```

Comment:
should be "manifest file node clean up"?

Reply:
Yes.